### DISCUSSION TITLE
Paramaters and useContractWrite

### DISCUSSION THREAD

**Discussion by isaac-martin** [wevm/wagmi#104] [Q&A] [answered]
Hey everyone - have used ethers in the past and am using this now. 
Not sure if this is me being thick, or something not working correctly. I couldn't figure it out from the docs nor looking through the src code. 
But how do you pass parameters into a function that you call using `useContratctWrite` ? I thought it would be on the args argument?

I used to have a `mint()` fn that just took a value - which worked fine. Have extended this function to take a qty as well and this is where my issue starts. The error i get is `Error: invalid BigNumber value (argument="value" .....`

TSX code looks like the below - will also put contract abi. 
```jsx

const contract = {...}

  const [{ data: mintData, error: mintError, loading: mintLoading }, mint] = useContractWrite(
  contract
   'mint');
  
  const doMint = () => {
    const mintCost = (mintQty * 0.1).toString()
    const cost = ethers.utils.parseEther(mintCost)


    mint({ args: { value: cost, _qty: mintQty } }
    );
  };
  
  ```
  
  ```json
   {
      "inputs": [
        {
          "internalType": "uint256",
          "name": "_qty",
          "type": "uint256"
        }
      ],
      "name": "mint",
      "outputs": [],
      "stateMutability": "payable",
      "type": "function"
    },
  ```

**Comment by isaac-martin** [comment, marked as answer]
Ok i got this working - but i think the types are wrong for overrides. 
This is the code that works to call the function defined above, having the value not as an arg (per this example https://github.com/tmm/wagmi/discussions/49#discussioncomment-1977456)  is what was throwing me - as that used to work when there were no other function arguments required 

<img width="943" alt="image" src="https://user-images.githubusercontent.com/8649366/150697144-5fff3430-9a28-426c-a913-fa0801de09dc.png">

Kind of looks like the types defined [here](https://github.com/tmm/wagmi/blob/main/packages/react/src/hooks/contracts/useContractWrite.ts#L11) - need to be `[PayableOverrides](https://github.com/ethers-io/ethers.js/blob/master/packages/contracts/src.ts/index.ts#L17)`
```ts
type Config = {
  args?: any | any[]
  overrides?: PayableOverrides
}
```

@tmm  happy to PR if you want to confirm thats the case - guess we could also add an optional field to the useContractWrite to indicate if its payable or not and adjust the types accordingly?

**Reply by tmm** [reply]
Fixed in https://github.com/tmm/wagmi/commit/b7018a64d88c9805a6fe0a4a41963ea6a33099a6
